osquery/tables/system/windows/windows_update_history.cpp

```diff
@@ -0,0 +1,381 @@
+        {WindowsUpdateHistoryError::CountError,
+         "Failed to get total update history count"},
+        {WindowsUpdateHistoryError::QueryError,
+         "Failed to query update history"},
+        {WindowsUpdateHistoryError::SizeError,
+         "Failed to get update history entry collection size"},
+        {WindowsUpdateHistoryError::EntryError,
+         "Failed to get update history entry"},
+        {WindowsUpdateHistoryError::ClientApplicationIDError,
+         "Failed to get history entry client application id"},
+        {WindowsUpdateHistoryError::DateError,
+         "Failed to get history entry date"},
+        {WindowsUpdateHistoryError::DescriptionError,
+         "Failed to get history entry description"},
+        {WindowsUpdateHistoryError::HResultError,
+         "Failed to get history entry HResult"},
+        {WindowsUpdateHistoryError::OperationError,
+         "Failed to get history entry operation"},
+        {WindowsUpdateHistoryError::ResultCodeError,
+         "Failed to get history entry result code"},
+        {WindowsUpdateHistoryError::ServerSelectionError,
+         "Failed to get history entry server selection"},
+        {WindowsUpdateHistoryError::ServiceIDError,
+         "Failed to get history entry service id"},
+        {WindowsUpdateHistoryError::SupportURLError,
+         "Failed to get history entry support URL"},
+        {WindowsUpdateHistoryError::TitleError,
+         "Failed to get history entry title"},
+        {WindowsUpdateHistoryError::IdentityError,
+         "Failed to get history entry identity"},
+        {WindowsUpdateHistoryError::UpdateIDError,
+         "Failed to get history entry update id"},
+        {WindowsUpdateHistoryError::UpdateRevisionError,
+         "Failed to get history entry update revision"},
+    };
+
+std::string getErrorDescription(const WindowsUpdateHistoryError& error) {
+  auto it = kWindowsUpdateHistoryErrorDescriptions.find(error);
+  if (it == kWindowsUpdateHistoryErrorDescriptions.end()) {
+    std::stringstream stream;
+    stream << "Unknown error type: 0x" << std::hex
+           << static_cast<std::uint64_t>(error);
+
+    return stream.str();
+  }
+
+  return it->second;
+}
+
+Expected<WindowsUpdateHistoryEntry, WindowsUpdateHistoryError>
+createWindowsUpdateHistoryEntryError(WindowsUpdateHistoryError error,
+                                     HRESULT hr) {
+  return createError(error)
+         << getErrorDescription(error) << ", HRESULT=0x" << std::hex << hr;
+}
+
+Expected<WindowsUpdateHistory, WindowsUpdateHistoryError>
+createWindowsUpdateHistoryError(WindowsUpdateHistoryError error, HRESULT hr) {
+  return createError(error)
+         << getErrorDescription(error) << ", HRESULT=0x" << std::hex << hr;
+}
+
+Row renderWindowsUpdateHistoryEntry(const WindowsUpdateHistoryEntry& entry) {
+  Row r;
+  r["client_app_id"] = entry.clientAppID;
+  r["date"] = BIGINT(entry.date);
+  r["description"] = BIGINT(entry.description);
```

Comment:
Is the `BIGINT` appropriate here? It seems `description` is a `std::string` and the table spec refers to it as `TEXT`

Reply:
yeah, this is a mistake, thanks for catching this, will update.